### ISSUE TITLE
wolfSSL Support

### ISSUE THREAD

**Issue by haydenroche5** [sudo-project/sudo#116]
Hi @millert  I'm a software engineer with wolfSSL, a lightweight TLS and cryptography library geared toward embedded systems. Additionally, we sell a commercial version of wolfSSL that's been FIPS-validated (https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/3389). We recently had a customer ask us to port sudo 1.9.5p2 so that it can use wolfSSL instead of OpenSSL or gcrypt. The patch for that work can be found here: https://github.com/wolfSSL/osp/pull/85/files.

Would you be interested in a pull request to add optional wolfSSL support for the latest sudo version? If so, I can adapt the 1.9.5p2 patch so that the changes apply cleanly onto the main branch. I just wanted to gauge interest before proceeding with a pull request. :) Thanks!

**Comment by millert** [comment]
Yes, please.  You can also just replace the X509_FILETYPE_PEM with SSL_FILETYPE_PEM in the call to SSL_CTX_use_PrivateKey_file() which reduces the diff slightly.
